Sources/Host/Common/Socket.cpp

```diff
@@ -355,10 +355,10 @@ std::string Socket::port() const {
     return std::string();
   }
 
+  auto func = listening() ? ::getsockname : ::getpeername;
   struct sockaddr_storage ss;
   socklen_t sslen = sizeof(ss);
-  if (::getsockname(_handle, reinterpret_cast<struct sockaddr *>(&ss), &sslen) <
-      0) {
+  if (func(_handle, reinterpret_cast<struct sockaddr *>(&ss), &sslen) < 0) {
```

Comment:
there's code duplication
I'd move lines 358, 360 and 361 to a new function, same as above, to follow the DRY rule